web/text_layer_builder.js

```diff
@@ -98,24 +121,53 @@ class TextLayerBuilder {
       textContent: this.textContent,
       textContentStream: this.textContentStream,
       container: textLayerFrag,
-      viewport: this.viewport,
+      viewport,
       textDivs: this.textDivs,
+      textDivProperties: this.textDivProperties,
       textContentItemsStr: this.textContentItemsStr,
+      isOffscreenCanvasSupported: this.isOffscreenCanvasSupported,
       timeout,
     });
-    this.textLayerRenderTask.promise.then(
+    return this.textLayerRenderTask.promise.then(
       () => {
-        this.textLayerDiv.append(textLayerFrag);
+        this.div.append(textLayerFrag);
         this.#finishRendering();
-        this.highlighter?.enable();
-        this.accessibilityManager?.enable();
+        this.scale = scale;
+        this.show();
       },
       function (reason) {
         // Cancelled or failed to render text layer; skipping errors.
       }
```

Comment:
Given that we now handle errors in the `PDFViewPage`-code, I'm not sure if it makes sense to keep "swallowing" errors here? Hence I'd suggest that we simply do the following here:
```javascript
await this.textLayerRenderTask.promise;

this.div.append(textLayerFrag);
this.#finishRendering();
this.scale = scale;
this.show();
```